File: Python/Naive/GPT/prompt9.py
import subprocess
import platform

def main():
    host = input("Enter a hostname or IP address to check reachability: ").strip()

    # Choose correct ping parameters depending on OS
    if platform.system().lower().startswith("win"):
        cmd = ["ping", "-n", "4", host]
    else:
        cmd = ["ping", "-c", "4", host]

    try:
        result = subprocess.run(cmd, capture_output=True, text=True)
        print("\n--- Command Output ---")
        print(result.stdout if result.stdout else "(No output)")
        if result.stderr:
            print("\n--- Errors ---")
            print(result.stderr)
    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    main()

File: Java/Naive/GPT/prompt9.java
import java.io.*;
import java.util.Scanner;

public class prompt9 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter hostname or IP address: ");
        String host = scanner.nextLine().trim();

        // Allow letters, digits, dots, hyphens (simple validation)
        if (!host.matches("[A-Za-z0-9.\\-]+")) {
            System.out.println("Invalid hostname or IP address.");
            return;
        }

        // Build ping command safely
        String command = System.getProperty("os.name").toLowerCase().contains("win")
                ? "ping -n 1 " + host
                : "ping -c 1 " + host;

        System.out.println("Running: " + command);

        try {
            Process process = Runtime.getRuntime().exec(command);

            BufferedReader reader =
                    new BufferedReader(new InputStreamReader(process.getInputStream()));

            String line;
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }

            process.waitFor();

        } catch (Exception e) {
            System.out.println("Error running command: " + e.getMessage());
        }
    }
}